### ISSUE TITLE
Any plans of making a binary release?

### ISSUE THREAD

**Issue by NonVolatileHuman** [kcat/openal-soft#961]

**Comment by kcat** [comment]
I'd need to look into how to make an installer for Windows that can put DLLs into global directories, with proper ways to update and uninstall them as needed. Though even with them installed, apps may need to be configured to use OpenAL Soft devices if it's not picked by default, which may or may not be easy depending on the app.

**Comment by NonVolatileHuman** [comment]
Yeah, I see your point, the advantages of the installer would be to eliminate friction in the installation, maybe in the future someone may contribute the necessary code. Thanks for taking the time and replying to this issue, you may close it if you see fit.

**Comment by TheBestKmanEver** [comment]
this would def be nicer than having to fumble about in the mess of all the different GUIs to build from source to get the files you need

**Comment by mirh** [comment]
> than having to use a CMake program

What does the build tool have to do with anything?

**Comment by NonVolatileHuman** [comment]
> What does the build tool have to do with anything?

Isn't it necessary to configure the tool for windows?
Or is there a more user-friendly way to install the software?

**Comment by mirh** [comment]
Cmake is used for compilation...
Configuration everywhere just uses [`alsoftrc.sample`](https://github.com/kcat/openal-soft/blob/master/alsoftrc.sample)

**Comment by ThreeDeeJay** [comment]
InnoSetup seems like a feasible option. I gather it just needs a script specifying what files to copy and where, among other things, like https://github.com/Energy0124/freepiano/compare/master...LJFloor:freepiano:master
And then it could be used in GitHub actions to compile and upload the installer to releases
https://github.com/noahsub/Orthographic-Renderer/commit/afb4839df6fe6d06eb01708d75aec948eb71a888#diff-006d566e109ce80a80cbd243c89a25b876792c89e18c2fb931eaba8535e4ed00R21-R27
It'd be neat to allow choosing between installing soft_oal.dll with/without the router, or just override it by renaming the library to OpenAL32.dll, optionally opening alsoft-config.exe after installation and/or adding a desktop shortcut, and adding preset configs like HRTF, surround, stereo, etc.

**Comment by mirh** [comment]
I wouldn't recommend installing it in place of the router, since it can be that games with the openal installer will overwrite it (I forgot whether they have any check whatsoever to avoid it).
On top of it.. Is system folder openal32.dll even picked up by those games that ship with their own?

**Comment by ThreeDeeJay** [comment]
Good question. I've never noticed any game not using the local OpenAL32.dll, if present (unlike dsound.dll) though I haven't been actively looking for it. I think the only ones that use the system ones are the ones that come with/install the redist to load it from the system folder.
That's why I personally just rather replace the game's OpenAL32.dll (or add it if it doesn't exist/expects to load the system one), to avoid any funny business going on in the system folders.

**Comment by mirh** [comment]
> replace the game's OpenAL32.dll

???
And how would an installer manage to do that? I didn't think it was possible to talk about anything but the normal one.

>  I think the only ones that use the system ones are the ones that come with/install the redist to load it from the system folder.

I could swear there was a (one) game that I had registered in my mind like "Mass Effect with physx, except for openal".
...but nonetheless most games aren't coded to use one or the other (an exception probably being [old ass](https://github.com/kcat/openal-soft/issues/1001#issuecomment-2308007137) ones?). It just depends on whether they have the dll in their binaries folder or not, and then loadlibrary priorities do the rest.

>  to avoid any funny business going on in the system folders.

There isn't really any funny combination or anything that I haven't already covered in, yet again, #757 and its forerunners.

**Comment by ThreeDeeJay** [comment]
> > replace the game's OpenAL32.dll
> 
> ???
> And how would an installer manage to do that? I didn't think it was possible to talk about anything but the normal one.

@mirh I meant just manually copying the file to the game folder

> It just depends on whether they have the dll in their binaries folder or not, and then loadlibrary priorities do the rest.

Yeah that's what i was referring to.

> There isn't really any funny combination or anything that I haven't already covered in, yet again, #757 and its forerunners.

I meant shenanigans like replacing the system router with OpenAL Soft, and then the redist reverting it like you mentioned. Replacing the local DLL would be isolated from unexpected system DLL changes, plus it'd update OpenAL Soft for games that come with it, which ignore  soft_oal.dll installed in the system folder AFAIK

**Comment by mirh** [comment]
I don't know what unexpected system dlls would even happen _ever_.. and I'm not sure what "it" might games already be coming with. The router? As I already covered, dll location should only matter there only as far as `ALC_ENUMERATE_ALL_EXT` is concerned (device iteration stops the moment they find whatever first partial match).
As for `ALC_ENUMERATION_EXT` I'm pretty sure regardless of the folder you cannot win unless you customize device or oal-soft names (or well, you delete wrap_oal).
